FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -128,19 +128,16 @@ final class ComposeMessageService {
     }
 
     // MARK: - Encrypt and Send
-    func encryptAndSend(message: SendableMsg, threadId: String?) -> AnyPublisher<Void, ComposeMessageError> {
-        return encryptMessage(with: message, threadId: threadId)
-            .flatMap(messageGateway.sendMail)
-            .mapError { ComposeMessageError.gatewayError($0) }
-            .eraseToAnyPublisher()
-    }
-
-    private func encryptMessage(with msg: SendableMsg, threadId: String?) -> AnyPublisher<MessageGatewayInput, Error> {
-        return core.composeEmail(
-            msg: msg,
-            fmt: MsgFmt.encryptInline
-        )
-        .map({ MessageGatewayInput(mime: $0.mimeEncoded, threadId: threadId) })
-        .eraseToAnyPublisher()
+    func encryptAndSend(message: SendableMsg, threadId: String?) async throws {
+        do {
+            let r = try await core.composeEmail(
+                msg: message,
+                fmt: MsgFmt.encryptInline
+            )
+
+            try await messageGateway.sendMail(input: MessageGatewayInput(mime: r.mimeEncoded, threadId: threadId))
+        } catch {
+            throw ComposeMessageError.gatewayError(error)
+        }
```

Comment:
This needs additional scrutiny in another issue. Error may be thrown by either composeEmail or sendMail. But either way we are throwing `gatewayError`. It should be balled `composeError` (and make sure it can handle both types of errors)